### PR TITLE
Require Node.js 6, add TypeScript definition

### PR DESCRIPTION
**Pull request by BendingBender**

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+	@param value - The value to resolve the promise with.
+	*/
+	resolve(value: T | PromiseLike<T>): void;
+
+	/**
+	Reject the promise with a provided reason or error.
+
+	@param reason - The reason or error to reject the promise with.
+	*/
+	reject(reason: any): void;
```
Comment (sindresorhus) [inline index.d.ts]:
```suggestion
	reject(reason: unknown): void;
```

File: index.d.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
+interface DeferredPromise<T> {
```
Comment (sindresorhus) [inline index.d.ts]:
```suggestion
export interface DeferredPromise<T> {
```

File: index.test-d.ts [hunk 3]
```diff
@@ -0,0 +1,9 @@
+import {expectType} from 'tsd-check';
+import pDefer, {DeferredPromise} from '.';
```
Comment (sindresorhus) [inline index.test-d.ts:2]: Why does it work to import `DeferredPromise` even though it's not exported?

File: index.d.ts [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+	Reject the promise with a provided reason or error.
+
+	@param reason - The reason or error to reject the promise with.
+	*/
+	reject(reason: any): void;
+
+	/**
+	The deferred promise.
+	*/
+	promise: Promise<T>;
```
Comment (sindresorhus) [inline index.d.ts]: Since `ValueType` is used earlier, we should use it here too for consistency.

```suggestion
	promise: Promise<ValueType>;
```

**Review by sindresorhus — CHANGES_REQUESTED** [review]

File: index.test-d.ts (hunk 3 above)
Comment (BendingBender) [reply index.test-d.ts:2]: You don't actually need to export interfaces fron an ambient declaration, they are exported automatically.

Comment (sindresorhus) [reply index.test-d.ts:2]: Ugh, that is not very good behavior... It should be explicit.

**Review by sindresorhus — APPROVED** [review]

File: index.test-d.ts (hunk 3 above)
Comment (BendingBender) [reply index.test-d.ts:2]: Yeah, forgot to export them...

Comment (sindresorhus) [reply index.test-d.ts:2]: To be clear, I was talking about the ambient file behavior, not you. We should really have a lint rule for this.

Comment (BendingBender) [reply index.test-d.ts:2]: `dts-lint` has a rule like this. Actually, `dts-lint` has most rules that you'd need to cover your type definition styleguide.

**Pull request merged**